src/llmq/signing.cpp

```diff
@@ -809,18 +828,38 @@ void CSigningManager::StopWorkerThread()
 void CSigningManager::InterruptWorkerThread()
 {
     workInterrupt();
+    workCv.notify_all();
 }
 
 void CSigningManager::WorkThreadMain(PeerManager& peerman)
 {
     while (!workInterrupt) {
+        uint64_t startEpoch = workEpoch.load(std::memory_order_acquire);
         bool fMoreWork = ProcessPendingRecoveredSigs(peerman);
 
         Cleanup();
 
-        // TODO Wakeup when pending signing is needed?
-        if (!fMoreWork && !workInterrupt.sleep_for(std::chrono::milliseconds(100))) {
-            return;
+        if (fMoreWork) continue;
+        std::unique_lock<Mutex> l(workMutex);
+        // Compute next cleanup deadline (~5s cadence) and wait event-driven until either
+        // new work arrives or the deadline is reached.
+        auto next_deadline = std::chrono::steady_clock::time_point::max();
```

Comment:
why not
```suggestion
        auto next_deadline{std::chrono::system_clock::now()};
```
and then...